pkg/k8shandler/visualization.go

```diff
@@ -36,146 +36,57 @@ var (
 // CreateOrUpdateVisualization reconciles visualization component for cluster logging
 func (clusterRequest *ClusterLoggingRequest) CreateOrUpdateVisualization(proxyConfig *configv1.Proxy) (err error) {
 	if clusterRequest.cluster.Spec.Visualization == nil || clusterRequest.cluster.Spec.Visualization.Type == "" {
-		if err = clusterRequest.removeKibana(); err != nil {
-			return
+		if err := clusterRequest.removeKibana(); err != nil {
+			logrus.Error(err.Error())
+			return nil
 		}
 		return nil
 	}
-	if clusterRequest.cluster.Spec.Visualization.Type == logging.VisualizationTypeKibana {
-
-		if err = clusterRequest.CreateOrUpdateServiceAccount(kibanaServiceAccountName, &kibanaServiceAccountAnnotations); err != nil {
-			return
-		}
 
-		if err = clusterRequest.RemoveClusterRoleBinding("kibana-proxy-oauth-delegator"); err != nil {
-			return
-		}
+	if err = clusterRequest.createKibanaCR(); err != nil {
+		logrus.Error(err.Error())
+		return nil
+	}
 
-		if err = clusterRequest.createOrUpdateKibanaService(); err != nil {
-			return
-		}
+	if err = clusterRequest.removeOldKibana(); err != nil {
```

Comment:
lets make this the first operation to reduce the risk of duplicate object names existing (if EO resolves before we complete clean up)

Reply:
Fixed